projects/igniteui-angular/src/lib/grids/grid/grid-validation.spec.ts

```diff
@@ -0,0 +1,405 @@
+            expect(cell.errorTooltip.first.collapsed).toBeFalse();
+        }));
+
+        it('should allow preventing edit mode for cell/row to end by canceling the related event if isValid event argument is false', () => {
+            const grid = fixture.componentInstance.grid as IgxGridComponent;
+            grid.cellEdit.pipe(takeUntil($destroyer)).subscribe((args) => {
+                if (!args.valid) {
+                    args.cancel = true;
+                }
+            });
+
+            let cell = grid.gridAPI.get_cell_by_visible_index(1, 1);
+            UIInteractions.simulateDoubleClickAndSelectEvent(cell.element);
+
+            const lastValue = cell.value;
+            cell.formControl.setValue('asd');
+            fixture.detectChanges();
+            grid.gridAPI.crudService.endEdit(true);
+            fixture.detectChanges();
+
+            cell = grid.gridAPI.get_cell_by_visible_index(1, 1);
+            // should have been canceled and left in editmode because of non valid value
+            // should not have updated the value
+            GridFunctions.verifyCellValid(cell, false);
+            expect(!!grid.gridAPI.crudService.rowInEditMode).toEqual(true);
+            expect(grid.gridAPI.crudService.cellInEditMode).toEqual(true);
+            expect(cell.value).toEqual(lastValue);
+
+            cell.update('test');
+            fixture.detectChanges();
+            cell = grid.gridAPI.get_cell_by_visible_index(1, 1);
+            GridFunctions.verifyCellValid(cell, true);
+        });
+
+        it('should trigger the validationStatusChange event on grid when validation status changes', () => {
+            const grid = fixture.componentInstance.grid as IgxGridComponent;
+            spyOn(grid.validationStatusChange, "emit").and.callThrough();
+
+            let cell = grid.gridAPI.get_cell_by_visible_index(1, 1);
+            UIInteractions.simulateDoubleClickAndSelectEvent(cell.element);
+            cell.editMode = true;
+            cell.update('asd');
+            fixture.detectChanges();
+            cell = grid.gridAPI.get_cell_by_visible_index(1, 1);
+
+            grid.crudService.endEdit(true);
+            fixture.detectChanges();
+
+            GridFunctions.verifyCellValid(cell, false);
+            expect(grid.validationStatusChange.emit).toHaveBeenCalledWith(Validity.Invalid);
+
+            UIInteractions.simulateDoubleClickAndSelectEvent(cell.element);
+            cell.editMode = true;
+            cell.update('test');
+            fixture.detectChanges();
+            cell = grid.gridAPI.get_cell_by_visible_index(1, 1);
+
+            grid.crudService.endEdit(true);
+            fixture.detectChanges();
+
+            GridFunctions.verifyCellValid(cell, true);
+            expect(grid.validationStatusChange.emit).toHaveBeenCalledWith(Validity.Valid);
+        });
+
+        xit('should return invalid transaction using the transaction`s getInvalidTransactionLog method', () => {
```

Comment:
Title can be updated a bit since this no longer tests  the transactions. Also it can be enabled now.